include/open62541/plugin/accesscontrol.h

```diff
@@ -80,27 +81,36 @@ struct UA_AccessControl {
     /* Allow adding a node */
     UA_Boolean (*allowAddNode)(UA_Server *server, UA_AccessControl *ac,
                                const UA_NodeId *sessionId, void *sessionContext,
-                               const UA_AddNodesItem *item);
+                               const UA_AddNodesItem *item, UA_RolePermissionType *userRolePermission, size_t userRoleSize);
```

Comment:
I think we can keep the current API intact.
By adding a lightweight method where we can look up the userRolePermissions from the session id.
So `allowNode` can read the permissions only when needed.